main/src/main/java/com/ldtteam/blockout/connector/common/inventory/provider/CommonEntityBasedProvider.java

```diff
@@ -81,27 +80,29 @@ public boolean equals(final Object o)
      */
     @NotNull
     @Override
-    public ResourceLocation getId()
+    public IIdentifier getId()
     {
-        return new ResourceLocation(id);
+        return IIdentifier.create(id);
     }
 
     /**
      *
-     * @return
      */
     @Nullable
     @Override
     public IItemHandler get(@NotNull final IItemHandlerManager manager)
     {
-        final World blockAccess = BlockOut.getBlockOut().getProxy().getWorldFromDimensionId(dimId);
-        final Entity entity = blockAccess.loadedEntityList.stream().filter(e -> e.getPersistentID().equals(entityId)).findFirst().orElse(null);
+        final IDimension<?> blockAccess = ProxyHolder.getInstance().getDimensionFromDimensionId(dimId);
+        final IEntity entity = blockAccess.getLoadedEntities().stream().filter(e -> e.getId().equals(entityId)).findFirst().orElse(null);
 
         if (entity == null)
         {
-            return new EmptyHandler();
+            return IItemHandler.empty();
         }
 
-        return entity.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, facing);
+        if (!entity.hasCapability(ICapability.getItemHandlerCapability(), facing))
+            throw new IllegalStateException("Gui created with an entity inventory for an entity that has no entity.");
```

Comment:
How does that exception message make any sense?

Reply:
Oh lol
It should say "for an entity that has no inventory on the given facing"....